datadog/tracing.go

```diff
@@ -0,0 +1,32 @@
+package datadog
+
+import (
+	"strconv"
+)
+
+const (
+	TraceIDKey = "dd.trace_id"
+	SpanIDKey  = "dd.span_id"
+)
+
+func TraceIDFromHexFormat(id string) string {
+	return convertHexIDToUint(id)
+}
+
+func SpanIDFromHexFormat(id string) string {
+	return convertHexIDToUint(id)
+}
+
+func convertHexIDToUint(id string) string {
+	if len(id) < 16 {
+		return ""
+	}
+	if len(id) > 16 {
+		id = id[16:]
+	}
+	intValue, err := strconv.ParseUint(id, 16, 64)
+	if err != nil {
+		return ""
+	}
+	return strconv.FormatUint(intValue, 10)
+}
```

Comment:
The only thing I'd change here is that when the string is not in a supported format, returning an empty string means we'll lose the trace/span ids from being logged.

WDYT if we return the not transformed `id`? Or maybe we could return a guard as well so the caller  can determine which logic to implement (return an empty string, an error, the id as it was, etc)

Reply:
> WDYT if we return the not transformed id?

I wouldn't return the original ID since DataDog expects it to be an integer (I couldn't find info on how they handle incorrect values).

> Or maybe we could return a guard as well so the caller can determine which logic to implement (return an empty string, an error, the id as it was, etc)

Couldn't consumers implement their own transformers to achieve this? IMHO leaving the dd trace_id/span_id is not such a big deal since most likely you'd also send the original traces as other tags

Reply:
Makes sense.
I didn't think that the library users could use both the `otel`-like option along with the datadog specialized one!